content/training/R_Basics/Advanced_Functions.Rmd

```diff
@@ -12,82 +12,99 @@ weight: 12
 
-If the file you want is in the folder designated as "here", all you have to do is provide the name of the file in quotes (with the appropriate ending). 
+```{r}
+# Use auto-generated dataset for this example
+linelist_cleaned <- gen_data("AJS")
 
-```{r eval=FALSE}
-linelist_raw <- rio::import(here("linelist.xlsx"), which = "Sheet1")
+# the variable is a factor. 
+class(linelist_cleaned$sex)
 ```
+*Note: To convert use as.factor() `linelist_cleaned$sex <- as.factor(linelist_cleaned$sex)`*  
 
-*Note: After the closing parenthes of here() the import() function continues. There is a comma and import()'s optional argument `which = `, specifying which sheet of the Excel file to import.*  
+**To change the labels for a variable, use the function fct_recode() from the package forcats.**  
 
-**If however, your file is within subfolders of the folder designated as "here"** - let's say it is in the subfolder "data" and then the sub-subfolder "linelists" - **you write these folder names before the file name, in quotes, and separated by commas**, as below:  
+After providing the name of the factor give the conversion statements, separated by commas, in the format **NEW = "OLD"**. Note that everything on the right-hand side (RHS) must be a character. So, if you want to give a new label to missing values, use the special term `NA_character_`.  
 
-```{r eval=FALSE}
-linelist_raw <- rio::import(here("data", "linelists", "linelist.xlsx"), which = "Sheet1")
+First we use fct_count() from the package forcats to see the values in the factor:
+
+```{r}
+# View the levels present in the variable
+fct_count(linelist_cleaned$sex)
 ```
 
-You can read more about Here and here() at [this site](https://github.com/jennybc/here_here) and [this one](https://here.r-lib.org/)  
+Now we re-label the values using fct_recode(), also from the package forcats.  
 
-*Note: by using the Here package and here() function with R projects, it is not necessary to worry about your "working directory", setwd(), or getwd() commands.*
+```{r}
+# Re-define the labels for the factor variable (pay attention to spelling!)
+linelist_cleaned$sex <- fct_recode(linelist_cleaned$sex, 
+                                   Male = "M",
+                                   Female = "F",
+                                   Unknown = "U")
+
+# View the levels present in the variable
+fct_count(linelist_cleaned$sex)
+```
 
-## Joins  
+**To change the *order* of the levels, use fct_relevel()**.  
 
-The package dplyr contains functions that are used in the templates for joining ("merging") data frames. 
-There are many options for joins, and it is important that you use the correct one - otherwise you risk losing data unexpectedly.
+This will impact the *order* of the levels as displayed in graphs and tables. Give the name of the factor followed by the new order (pay attention to spelling!).  
```

Comment:
```suggestion
This will impact the *arrangement* of the levels as displayed in graphs and tables. Give the name of the factor followed by the new arrangement (pay attention to spelling!).  
```